src/uct/tcp/tcp_net.c

```diff
@@ -26,12 +26,49 @@
 
 typedef ssize_t (*uct_tcp_io_func_t)(int fd, void *data, size_t size, int flags);
 
+/* Caller is responsible to free memory allocated if str_addr wasn't provided */
+char *uct_tcp_sockaddr_2_string(const struct sockaddr_in *addr, char **str_addr,
+                                size_t *str_addr_len)
+{
+    int ret = 0;
+
+    if (str_addr != NULL && *str_addr != NULL && str_addr_len && NULL) {
+        ret = snprintf(*str_addr, *str_addr_len, "%s:%d",
```

Comment:
potential issue: snprintf returns length required to print the whole line, it could be longer that *str_addr_len

Reply:
good catch! that was not my idea. thank you!
fixed.